tests/common/user_test.go

```diff
@@ -0,0 +1,371 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.etcd.io/etcd/tests/v3/framework/config"
+	"go.etcd.io/etcd/tests/v3/framework/testutils"
+)
+
+func TestUserAdd_Simple(t *testing.T) {
+	testRunner.BeforeTest(t)
+	tcs := []struct {
+		name   string
+		config config.ClusterConfig
+	}{
+		{
+			name:   "NoTLS",
+			config: config.ClusterConfig{ClusterSize: 1},
+		},
+		{
+			name:   "PeerTLS",
+			config: config.ClusterConfig{ClusterSize: 3, PeerTLS: config.ManualTLS},
+		},
+		{
+			name:   "PeerAutoTLS",
+			config: config.ClusterConfig{ClusterSize: 3, PeerTLS: config.AutoTLS},
+		},
+		{
+			name:   "ClientTLS",
+			config: config.ClusterConfig{ClusterSize: 1, ClientTLS: config.ManualTLS},
+		},
+		{
+			name:   "ClientAutoTLS",
+			config: config.ClusterConfig{ClusterSize: 1, ClientTLS: config.AutoTLS},
+		},
+	}
+	for _, tc := range tcs {
+		nestedCases := []struct {
+			name          string
+			username      string
+			password      string
+			noPassword    bool
+			expectError   bool
+			expectedError string
+		}{
+			{
+				name:        "empty_username_not_allowed",
+				username:    "",
+				password:    "foobar",
+				expectError: true,
+				// Very Vague error expectation because the CLI and the API return very
+				// different error structures.
+				expectedError: "user name",
+			},
+			{
+				// Can create a user with no password, restricted to CN auth
+				name:        "no_password_with_noPassword_set",
+				username:    "foo",
+				password:    "",
+				noPassword:  true,
+				expectError: false,
+			},
+			{
+				// Can create a user with no password, but not restricted to CN auth
+				name:        "no_password_without_noPassword_set",
+				username:    "foo",
+				password:    "",
+				noPassword:  false,
+				expectError: false,
```

Comment:
A non-empty password is always required when `NoPassword` is false, so the integration test should fail in this case. Please double check.